sparta.go

```diff
@@ -933,6 +943,20 @@ func NewLambda(roleNameOrIAMRoleDefinition interface{},
 	return lambda
 }
 
+// NewNamedLambda returns a LambdaAWSInfo value with a declared function name that can be provisioned
+// via CloudFormation. The `functionName` value must be a non-empty string, otherwise CloudFormation
+// will error out as "" is not a valid function name. The roleNameOrIAMRoleDefinition must either be
+// a `string` or `IAMRoleDefinition` type
+func NewNamedLambda(roleNameOrIAMRoleDefinition interface{},
+fn LambdaFunction,
+lambdaOptions *LambdaFunctionOptions,
+functionName string) *LambdaAWSInfo {
```

Comment:
Thoughts on changing this to `(*LambdaAWSInfo, error)` to catch empty `functionName` values prior to the CloudFormation marshalling step?